test/cppsuite/tests/hs_cleanup.cxx

```diff
@@ -28,31 +28,69 @@
+class hs_cleanup : public test {
     public:
     hs_cleanup(const std::string &config, const std::string &name) : test(config, name) {}
 
     void
-    populate(test_harness::database &database, test_harness::timestamp_manager *_timestamp_manager,
-      test_harness::configuration *_config, test_harness::workload_tracking *tracking)
-       override final
+    update_operation(thread_context *tc) override final
     {
-        std::cout << "populate: nothing done." << std::endl;
-    }
+        WT_CURSOR *cursor = nullptr;
+        WT_DECL_RET;
+        const char *key_tmp;
+        WT_SESSION *session = connection_manager::instance().create_session();
+        std::string collection_name = tc->database.get_collection_name(tc->id);
+        wt_timestamp_t ts;
 
-    void
-    read_operation(test_harness::thread_context *context) override final
-    {
-        std::cout << "read_operation: nothing done." << std::endl;
-    }
+        /* In this test each thread gets a single collection. */
+        testutil_assert(tc->database.get_collection_count() == tc->thread_count);
+        testutil_check(session->open_cursor(session,  collection_name.c_str(), nullptr, nullptr, &cursor));
 
-    void
-    update_operation(test_harness::thread_context *context) override final
-    {
-        std::cout << "update_operation: nothing done." << std::endl;
+        /* We don't know the keyrange we're operating over here so we can't be much smarter here. */
+        while (tc->running()) {
+            tc->sleep();
+            ret = cursor->next(cursor);
+            if (ret != 0) {
+                if (ret == WT_NOTFOUND) {
+                    testutil_check(cursor->reset(cursor));
+                    continue;
+                } else
+                    testutil_die(ret, "cursor->next() failed unexpectedly.");
+            }
+            testutil_check(cursor->get_key(cursor, &key_tmp));
+
+            /* Start a transaction. */
+            if (!tc->transaction.active())
+                tc->transaction.begin(tc->session, "");
+
+            ts = tc->timestamp_manager->get_next_ts();
+            if (tc->timestamp_manager->enabled())
+                tc->transaction.set_commit_timestamp(
+                  tc->session, timestamp_manager::decimal_to_hex(ts));
+
+            /* Update the record but take care to handle WT_ROLLBACK. */
+            ret = update(tc->tracking, cursor, collection_name,
+              key_value_t(key_tmp), random_generator::instance().generate_string(tc->value_size).c_str(), ts);
+            /* Increment the current op count for the current transaction. */
+            tc->transaction.op_count++;
+            if (ret == WT_ROLLBACK)
+                tc->transaction.rollback(tc->session, "");
+            else if (ret != 0)
+                testutil_die(ret, "failed to update a key.");
+            if (tc->transaction.can_commit())
+                tc->transaction.commit(tc->session, "");
+        }
```

Comment:
As in the default implementation, you need to check that the last operation is committed:
`        if (tc->transaction.active())
            tc->transaction.commit(tc->session, "");`